### PR TITLE
feat(debug): cache, install, network, appium, provenance & findings sections for the diagnostic dump

### PR DESCRIPTION
**Pull request by hawkeyexl**
## What

Follow-up enhancements to the `doc-detective debug` diagnostic dump (introduced in #336). Adds six additive, read-only sections that target the install/cache complexity the dump exists to debug, plus an interpretation layer that turns the data into fix commands.

Each section is a typed `DebugData` field rendered through the existing `renderText()` → `Section[]` pipeline, so it also serializes into the `debug-<ts>.json` dump for free. Every collector is crash-proof (degrades to an `error`/`<unknown>` marker) and routes secret-bearing values through the existing `redactValue`/`redactObject` redaction layer.

### Sections

| # | Section | What it shows |
|---|---------|---------------|
| #2 | **Install status** | The installer's `status()` matrix — npm packages + browsers, installed vs expected vs `OUTDATED`, and the `installed.json` path. |
| #1 | **Cache & runtime** | Resolved `cacheDir` / `runtimeDir` / `browsersDir` / `installed.json` / `APPIUM_HOME` with exists, **writable**, and free-disk-space per location. |
| #6 | **Proxy / npm network** | Redacted `HTTP(S)_PROXY` / `NO_PROXY` + swept `npm_config_*` block — the #1 silent lazy-install failure mode. |
| #5 | **Config provenance** | Config file path, whether `DOC_DETECTIVE_CONFIG` / `DOC_DETECTIVE_API` applied, and which CLI flags overrode the validated config. |
| #4 | **Appium registration** | Reconciles npm-resolvable drivers against Appium's *registered* drivers (read from `extensions.yaml`): `registered` / `resolvable but not registered` / `missing`. |
| #3 | **Findings / next steps** | Pure rules over the collected `DebugData` that emit actionable fixes, rendered **first** as the support summary. |

## Why

The dump described the environment but stopped short of the two things that make install/cache bugs actually diagnosable: the resolved cache/runtime paths + install-status matrix the lazy installer works against, and an interpretation layer that says *what to do*. A user filing a bug can now paste one block that not only shows the state but flags the likely cause and the fix command (e.g. "Chrome unavailable → `doc-detective install runtime`"; "cacheDir not writable → set `DOC_DETECTIVE_CACHE_DIR`").

## Notes for reviewers

- **No schema changes** — this is all in `src/debug/*`. The `src/common` AJV schemas/types are untouched.
- **Config provenance keeps `setConfig` untouched.** Provenance is derived in the debug layer from inputs the command already has (configPath, `process.env`, parsed `args`), rather than threading new state out of `setConfig`. `args` is now passed into `printDebug` from all three call sites (the `debug` subcommand + both `DOC_DETECTIVE_DEBUG` paths in `cli.ts`).
- **Appium uses no subprocess.** The first revision spawned `appium driver list --installed`, but that command cold-loads Appium and does a network update-check (measured ~8.8s cold vs ~0.6s warm), so a bounded probe intermittently timed out and then mislabeled every registered driver as "not registered". It now reads `extensions.yaml` — Appium's own authoritative manifest, matched by `pkgName` — which is instant, offline, and deterministic. When the manifest is absent/unreadable, registration is reported as *unknown* rather than falsely "not registered", and the corresponding finding is suppressed.
- **Out of scope (own PRs):** the opt-in network-reachability probe (`--check-network`, #7) and browser-launch smoke test (`--probe-browser`, #8) — both have egress/process side effects.
- This is a `feat:`, so it will cut a **minor** release on merge.

## Testing

- `test/debug.test.js`: **71 passing**, 3 POSIX-only skips on Windows — covers each new collector/helper, the findings rules (including a regression test that a missing Appium manifest must not false-flag drivers), and an end-to-end section-presence + ordering check.
- Full main suite green except 7 pre-existing environmental failures (the documented port-8092 test-server collision in `core-screenshot` / `core-core` URL-fixture tests — unrelated to `src/debug`).
- Manually verified the live CLI: all sections render in order (Findings first), proxy/registry credentials are redacted (0 leaks), provenance reflects CLI overrides, and the Appium section runs in a steady ~770ms with drivers correctly shown as `registered`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Enhanced diagnostic output with comprehensive system, installation, cache, and network configuration details.
  * Added automated findings and recommendations based on diagnostic analysis to help troubleshoot common issues.
  * Improved debug dumps to include command-line argument provenance and configuration override tracking.

* **Tests**
  * Added comprehensive test coverage for new diagnostic modules and debug output functionality.

### CODE REVIEW

File: src/debug/provenance.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  {
+    flag: "test",
+    configKey: "testFilter",
+    present: (a) => typeof a.test === "string" && a.test.length > 0,
+  },
+  {
+    flag: "spec",
+    configKey: "specFilter",
+    present: (a) => typeof a.spec === "string" && a.spec.length > 0,
+  },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/provenance.ts]: Fixed in 76d35bb3 — same commaListPresent() guard applied to --spec.

File: src/debug/provenance.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  args?: any;
+  env?: NodeJS.ProcessEnv;
+}): Provenance {
+  const env = opts.env ?? process.env;
+  return {
+    configPath: opts.configPath ?? null,
+    docDetectiveConfigApplied: typeof env.DOC_DETECTIVE_CONFIG === "string",
+    docDetectiveApiApplied: typeof env.DOC_DETECTIVE_API === "string",
+    cliOverrides: collectCliOverrides(opts.args),
+  };
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/provenance.ts:157]: Fixed in 76d35bb3 — now uses Boolean(env.X) so an empty-string DOC_DETECTIVE_CONFIG/DOC_DETECTIVE_API reports 'not set', matching the if(!process.env.*) env readers.

File: src/debug/cache.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    const probe = safeExists(target) ? target : path.dirname(target);
+    const stats = statfsSync(probe) as { bavail?: number; bsize?: number };
+    if (typeof stats?.bavail !== "number" || typeof stats?.bsize !== "number") {
+      return null;
+    }
+    return stats.bavail * stats.bsize;
+  } catch {
+    return null;
+  }
+}
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/cache.ts:96]: Fixed in 76d35bb3 — freeSpaceBytes() uses the same nearestExistingPath() ancestor walk, so a deep not-yet-created path no longer yields a spurious <unknown>.

File: src/debug/provenance.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+// documented precedence chain (file → env → AJV → CLI) in setConfig
+// (src/utils.ts) WITHOUT threading new state out of setConfig: everything
+// here is derived from inputs the debug command already has, which keeps
+// setConfig untouched and this logic pure / unit-testable.
+
+export interface CliOverride {
+  // The CLI flag the user passed (without leading dashes).
+  flag: string;
+  // The config key it overrides after validation.
+  configKey: string;
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/provenance.ts:16]: Fixed in 76d35bb3 — flag now reports the real CLI spelling (--dry-run, --cache-dir, --allow-unsafe, …) via a separate argKey for the camel argv lookup; comment updated.

File: src/debug/provenance.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    configKey: "allowUnsafeSteps",
+    present: (a) => typeof a.allowUnsafe === "boolean",
+  },
+  { flag: "dryRun", configKey: "dryRun", present: (a) => typeof a.dryRun === "boolean" },
+  { flag: "reporters", configKey: "reporters", present: (a) => a.reporters != null },
+  {
+    flag: "test",
+    configKey: "testFilter",
+    present: (a) => typeof a.test === "string" && a.test.length > 0,
+  },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/provenance.ts]: Fixed in 76d35bb3 — commaListPresent() now comma-splits/trims/drops-empties like setConfig, so ',' or whitespace-only --test no longer reports an override.

File: src/debug/provenance.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  { flag: "input", configKey: "input", present: (a) => Boolean(a.input) },
+  { flag: "output", configKey: "output", present: (a) => Boolean(a.output) },
+  { flag: "logLevel", configKey: "logLevel", present: (a) => Boolean(a.logLevel) },
+  {
+    flag: "allowUnsafe",
+    configKey: "allowUnsafeSteps",
+    present: (a) => typeof a.allowUnsafe === "boolean",
+  },
+  { flag: "dryRun", configKey: "dryRun", present: (a) => typeof a.dryRun === "boolean" },
+  { flag: "reporters", configKey: "reporters", present: (a) => a.reporters != null },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/provenance.ts]: Fixed in 76d35bb3 — reportersPresent() now mirrors setConfig: reporters:[] (empty normalized list) no longer reports an override.

File: src/debug/cache.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+// is answered even for not-yet-created dirs.
+function isWritable(target: string): boolean {
+  try {
+    const probe = safeExists(target) ? target : path.dirname(target);
+    fs.accessSync(probe, fs.constants.W_OK);
+    return true;
+  } catch {
+    return false;
+  }
+}
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/cache.ts:77]: Fixed in 76d35bb3 — isWritable() now walks up to the nearest existing ancestor (nearestExistingPath) instead of a single dirname() step.

**Comment by hawkeyexl** [comment]
Pushed 97d0afe4 addressing the second review pass (prose-only, no inline threads):

- **findings.ts** — `chromeUnavailable` detail now distinguishes *missing* vs *installed-but-unregistered* vs *registration unknown* (manifest absent → `registered: null`), so the message is accurate when the driver package resolves but registration is merely unverified. Pinned with a test.
- **install.ts (new)** — moved `InstallData` + `collectInstallStatus` out of `index.ts` into their own module, matching the one-type-per-module pattern of the other collectors.
- **provenance.ts** — strengthened the `OVERRIDE_SPECS` sync note (the pinned-list test doesn't cross-reference `setConfig`, so a new override needs an entry here *and* a test update).
- **network.ts** — documented the case-insensitive `npm_config_*` prefix match and why uppercase `_authToken` variants still redact.

Intentionally left as-is: the duplicate `setAppiumHome(ctx)` call across `collectCacheStatus` / `collectAppiumDiagnostics` — it's idempotent and cheap, and keeping it in each collector lets them run correctly standalone (the unit tests call each directly). Debug suite: 75 passing, 0 failing.

**Comment by hawkeyexl** [comment]
Pushed d1ad0cc7 for the third pass (prose-only, no inline threads):

- **findings.ts — proxy detection gap**: `proxyMaybeBlocking` now also matches an npm-only proxy (`npm_config_proxy` / `npm_config_https_proxy`), so a proxy set via `npm config set proxy=…` no longer silently skips the finding. Closed the gap rather than just commenting it; pinned with a test.
- **cache.ts / appium.ts — duplicate setAppiumHome**: added comments at both call sites clarifying that each collector resolves `APPIUM_HOME` independently (idempotent, order-independent), so the second call in a full dump is a no-op, not a hidden ordering dependency. Kept the per-collector calls so each is correct standalone (the unit tests exercise them directly).

**Declined (with reason): the `reportersPresent([""])` suggestion.** I checked `setConfig` (src/utils.ts:366–373) — it overrides on `reporterList.length > 0` and does **not** drop empty-string entries (unlike `--test`/`--spec`). So `reporters: [""]` *does* override (sets `config.reporters = [""]`). My `reportersPresent` returns `true` for `[""]`, which faithfully mirrors that. Applying the suggested `.filter(r => r.length>0)` would make provenance report "no override" while `setConfig` actually overrides — a divergence. Added a comment documenting why empties are intentionally not filtered here.

Debug suite: 76 passing, 0 failing.

**Comment by hawkeyexl** [comment]
Pushed 80935064 closing the last guard asymmetry from the fourth-pass review:

- **provenance.ts** — added `concurrentRunnersPresent()` mirroring setConfig's `--concurrent-runners` validity gate (`""` / `"true"` / positive integer override; `"abc"`/`"0"` log a warning and skip). Provenance no longer reports an override for invalid values. Pinned with edge-case tests.

The other fourth-pass nit (`proxyMaybeBlocking` only checks `kind === "npm"` rows, not browser binaries) is intentional and matches the PR's documented scope — reachability/browser-download checks are deferred to the opt-in `--check-network` probe (#7). Leaving as-is.

With this, all four guards (`commaListPresent`, `reportersPresent`, `concurrentRunnersPresent`, plus the boolean/length checks) faithfully mirror their setConfig counterparts. Debug suite: 76 passing, 0 failing. The fourth review marked the PR ready to merge.

File: test/debug.test.js [hunk 8]
```diff
@@ -990,3 +990,455 @@ describe("debug CLI smoke test", function () {
...
+    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), "dd-cache-ro-"));
+    try {
+      fs.chmodSync(tmp, 0o500); // r-x, no write
+      const status = collectCacheStatus({ cacheDir: tmp });
+      const cacheDir = status.entries.find((e) => e.label === "cacheDir");
+      expect(cacheDir.writable).to.equal(false);
+    } finally {
+      fs.chmodSync(tmp, 0o700);
+      fs.rmSync(tmp, { recursive: true, force: true });
+    }
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/debug.test.js:1234]: Good catch — fixed in 5aa3395c. The test now also skips when process.getuid() === 0, since root bypasses the directory write-bit (a 0500 dir still passes W_OK), which would falsely fail the writable=false assertion in CI containers.

File: src/debug/cache.ts [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    // installed.json is a file, not a dir — report existence + whether its
+    // parent is writable (so "can the installer record here?" is answered),
+    // but no free-space figure.
+    const recordPath = getInstalledRecordPath(ctx);
+    entries.push({
+      label: "installed.json",
+      path: recordPath,
+      exists: safeExists(recordPath),
+      writable: isWritable(recordPath),
+      freeBytes: null,
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/cache.ts:126]: Fixed in f6f567b8 — installed.json now probes isWritable(path.dirname(recordPath)). writeInstalledRecord() uses tmp-file + rename, so directory writability is the right signal; checking the file path was wrong when a read-only file sat in a writable dir (or vice versa).

File: src/debug/cache.ts [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+    });
+
+    // APPIUM_HOME is resolved by mutating process.env (setAppiumHome returns
+    // void), then read back. May be left unset if nothing resolves. This is
+    // idempotent and resolved independently here (the appium collector calls
+    // it too) so each collector is correct standalone and in any order — the
+    // second call in a full dump is a cheap no-op, not an ordering dependency.
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/cache.ts]: Fixed in f6f567b8 — corrected the comment: setAppiumHome always assigns APPIUM_HOME (its legacy fallback is unconditional), so the unset branch only fires if the call threw and was swallowed above.

File: src/debug/index.ts [hunk 11]
```diff
@@ -416,17 +469,167 @@ function writeFileSafe(
...
+    // unknown, not confirmed-absent.
+    const classification =
+      d.registered === true
+        ? "registered"
+        : !d.npmResolvable
+        ? "missing"
+        : d.registered === false
+        ? "resolvable but not registered"
+        : "resolvable (registration unknown — no manifest)";
+    lines.push(`    ${d.name.padEnd(24)} ${classification}`);
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/index.ts:580]: Fixed in f6f567b8 — the unknown-registration classification now distinguishes 'manifest unreadable' (manifestError set) from 'no manifest' (absent), instead of hard-coding 'no manifest' for both null cases.

File: src/debug/network.ts [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    if (typeof env[key] === "string") names.add(key);
+  }
+  // Sweep every npm_config_* var present (registry, proxy, strict-ssl,
+  // cafile, per-registry _authToken, …). The prefix match is lowercased so
+  // uppercase variants (NPM_CONFIG_*) are caught too; keys keep their original
+  // casing, and redactValue/isSecretName are case-insensitive, so an
+  // _authToken key still redacts. Redaction handles the auth ones.
+  for (const key of Object.keys(env)) {
+    if (key.toLowerCase().startsWith("npm_config_")) names.add(key);
+  }
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/debug/network.ts:59]: Fixed in 46d8462a — the npm_config_* sweep now guards on typeof env[key] === "string" (matching the proxy loop), so a key present with an undefined value no longer surfaces as a misleading npm_config_x = <unset>. Pinned with a test.
